crates/compiler/solve/src/solve.rs

```diff
@@ -1771,6 +1771,66 @@ fn solve(
 
                 state
             }
+            IngestedFile(type_index, file_path, bytes) => {
+                let actual = either_type_index_to_var(
+                    subs,
+                    rank,
+                    pools,
+                    problems,
+                    abilities_store,
+                    obligation_cache,
+                    &mut can_types,
+                    aliases,
+                    *type_index,
+                );
+
+                let snapshot = subs.snapshot();
+                if let Success { .. } = unify(
+                    &mut UEnv::new(subs),
+                    actual,
+                    Variable::LIST_U8,
+                    Mode::EQ,
+                    Polarity::OF_VALUE,
+                ) {
+                    // List U8 always valid.
+                    subs.rollback_to(snapshot);
```

Comment:
I don't think we can rollback here. For example, someone might do

```
"my file.txt" as ownCode : _,
```

if this is rolled back, then `ownCode` is unbound in its type - but in fact it should be bound to `List U8`.

Reply:
Ok. Yeah, I was able to make an example where the type hole could break things. Fixed. Also added the test case as another cli run example. We should be good now.